packages/core/logging/core-logging-server-internal/src/logging_system.test.mocks.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { merge, getFlattenedObject } from '@kbn/std';
+
+export const mockStreamWrite = jest.fn();
+jest.doMock('fs', () => ({
+  ...(jest.requireActual('fs') as any),
+  constants: {},
+  createWriteStream: jest.fn(() => ({ write: mockStreamWrite })),
+}));
+
+export const mockGetFlattenedObject = jest.fn().mockImplementation(getFlattenedObject);
+jest.doMock('@kbn/std', () => ({
+  merge: jest.fn().mockImplementation(merge),
+  getFlattenedObject: mockGetFlattenedObject,
+}));
```

Comment:
nit: we probably can achieve the same behaviour with `jest.spyOn`?

Reply:
I started with that approach, but ran into [this issue](https://stackoverflow.com/a/69951703), which is why I reverted to using `jest.mock`